parsl/tests/test_shutdown/test_kill_monitoring.py

```diff
@@ -0,0 +1,65 @@
+import os
+import parsl
+import pytest
+import signal
+import time
+
+from parsl.tests.configs.htex_local_alternate import fresh_config
+
+# This is a very generous upper bound on expected shutdown time of target
+# process after receiving a signal, measured in seconds.
+PERMITTED_SHUTDOWN_TIME_S = 60
+
+
+@parsl.python_app
+def simple_app():
+    return True
+
+
+@pytest.mark.local
+def test_no_kills():
+    """This tests that we can create a monitoring-enabled DFK and shut it down."""
+
+    parsl.load(fresh_config())
+
+    assert parsl.dfk().monitoring is not None, "This test requires monitoring"
+
+    parsl.dfk().cleanup()
+    parsl.clear()
+
+
+@pytest.mark.local
+@pytest.mark.parametrize("sig", [signal.SIGINT, signal.SIGTERM, signal.SIGKILL])  # are we expecting SIGKILL resilience here? Ideally yes
```

Comment:
If resilient to `SIGKILL` (per the comment which can maybe go or be moved to a more appropriate place?), then the other signals almost aren't necessary.  Meanwhile, are any other signals of interest to verify?  `SIGHUP` and `SIGQUIT` come to mind.  (`SIGHUP` less from a "what it should do" perspective and more from a "what it _does do_ perspective.)

Reply:
I feel like if the code is installing INT or TERM handlers (which someone might do) there is opportunity for shutdown mischief that is not available with SIGKILL (for example, it can go peturb the shutdown behaviour of other components...) so should still be tested.

SIGQUIT I guess should behave mostly like KILL in Python, I think, so I'll add that and inevitably discover it's not true.

SIGHUP is interpreted sufficiently weirdly by daemons conventionally (eg re-reading config files is one convention) that I wouldn't expect any particular behaviour here...

Reply:
I've added SIGQUIT. The Philosophy of SIGHUP will have to wait 'till another day.